assert/assertions_test.go

```diff
@@ -2786,6 +2786,18 @@ func TestEventuallyWithTTrue(t *testing.T) {
 	Len(t, mockT.errors, 0)
 }
 
+func TestEventuallyWithT_ConcurrencySafe(t *testing.T) {
+	mockT := new(CollectT)
```

Comment:
Don't use `CollectT` here.

Reply:
Done. Please note this prevents us from checking how many errors were reported in this test. Nevertheless we have others that do check that and this one was meant for triggering the race itself so I guess it can be like that.

Reply:
I didn't say to not capture errors.

Just don't use CollectT for that.

Because my aim is to deprecate CollectT.

Reply:
👍 Added `errorsCapturingT` mock implementation so that it fits the `EventuallyWithT` needs.